spec/mouse.spec.js

```diff
@@ -184,83 +185,110 @@ describe('handling mouse in static objects', function() {
-                    target.unbind(event_type, trg_handlers[event_type]);
-                }
-            }
+            log.unsubscribe();
+            log.clear();
         });
 
         /* it('transfers in/out event to the corresponding receivers', function() {
             expect({ type: 'mousemove', x: 25, y: 75 })
                 .toBeHandledAs([ { in: e2,   target: e2, type: 'mouseenter', x: 25, y: 30 },
                                  { in: root, target: e2, type: 'mouseenter', x: 25, y: 30 } ]);
+
+            fireCanvasEvent('mousemove', 10, 10);
         }); */
 
         it('in/out events properly work in sequences', function() {
 
             fireCanvasEvent('mousemove', 25, 75);
-            expect(log.join(MARKER)).toEqual('root: mouseenter@25;30 -> e2' + MARKER +
-                                             'e2: mouseenter@25;30 -> e2');
+            expect(log.stringify(MARKER)).toEqual([ 'root: mouseenter@25;30 -> e2',
+                                                    'e2: mouseenter@25;30 -> e2' ].join(MARKER));
 
-            log = [];
+            log.clear();
 
             fireCanvasEvent('mousemove', 25, 75);
-            expect(log.join(MARKER)).toEqual('');
+            expect(log.stringify(MARKER)).toEqual('');
 
             fireCanvasEvent('mousemove', 26, 76);
-            expect(log.join(MARKER)).toEqual('');
+            expect(log.stringify(MARKER)).toEqual('');
 
             fireCanvasEvent('mousemove', 25, 25);
-            expect(log.join(MARKER)).toEqual('e2: mouseexit@null;null -> e2' + MARKER +
-                                             'e1: mouseenter@25;25 -> e11' + MARKER +
-                                             'e11: mouseenter@25;25 -> e11');
+            expect(log.stringify(MARKER)).toEqual([ 'e2: mouseexit@null;null -> e2',
+                                                    'e1: mouseenter@25;25 -> e11',
+                                                    'e11: mouseenter@25;25 -> e11' ].join(MARKER));
 
-            log = [];
+            log.clear();
 
             fireCanvasEvent('mousemove', 76, 6);
-            expect(log.join(MARKER)).toEqual('e11: mouseexit@null;null -> e11' + MARKER +
-                                             'e12: mouseenter@1;1 -> e12');
+            expect(log.stringify(MARKER)).toEqual([ 'e11: mouseexit@null;null -> e11',
+                                                    'e12: mouseenter@1;1 -> e12' ].join(MARKER));
 
-            log = [];
+            log.clear();
 
             fireCanvasEvent('mousemove', 25, 75);
-            expect(log.join(MARKER)).toEqual('e12: mouseexit@null;null -> e12' + MARKER +
-                                             'e1: mouseexit@null;null -> e12' + MARKER +
-                                             'e2: mouseenter@25;30 -> e2');
+            expect(log.stringify(MARKER)).toEqual([ 'e12: mouseexit@null;null -> e12',
+                                                    'e1: mouseexit@null;null -> e12',
```

Comment:
Because user may be interested where is the mouse position of first enter, or else she should extract it from mousemove and determine which mousemove was first. On exit, position is not important and it's hard to determine which element it should be relative to, though we could have both for consistency.